### PR TITLE
Merging Abhi's Raingage API work

### LINKED ISSUE #199: Add support to RainGage API

### PR DESCRIPTION
**Pull request by bemcdonnell**
Closes #155 
Closes #199

### CODE REVIEW

**Comment by bemcdonnell** [comment]
@abhiramm7, Have a look at this PR when you get a second.  I think this is ready to go

File: pyswmm/tests/data/__init__.py [hunk 1]
```diff
@@ -21,6 +21,9 @@
 MODEL_STORAGE_PUMP = os.path.join(DATA_PATH, 'model_storage_pump.inp')
 MODEL_STORAGE_PUMP_MGD = os.path.join(DATA_PATH, 'model_storage_pump_MGD.inp')
 MODEL_POLLUTANTS_PATH = os.path.join(DATA_PATH, 'model_pollutants.inp')
+MODEL_RAIN = os.path.join(DATA_PATH, 'model_rain.inp')
+WIN_SWMM_LIB_PATH = os.path.join(DATA_PATH, '..\\..\\lib\\windows',
```
Comment (jennwuu) [inline pyswmm/tests/data/__init__.py]: WIN_SWMM_LIB_PATH is set in lines 27-34

File: pyswmm/tests/test_rainfallAPI.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+from pyswmm import Simulation, RainGages, SystemStats
+from pyswmm.tests.data import MODEL_RAIN
+
+def test_api_rainfall():
+    with Simulation(MODEL_RAIN) as sim:
+        check = 0
+
+        sim._model.setGagePrecip("Gage1", 10.00)
+        for ind, step in enumerate(sim):
+            sim._model.setGagePrecip("Gage1", 10.00)
```
Comment (jennwuu) [inline pyswmm/tests/test_rainfallAPI.py]: can rain gage precip be set during model simulation

File: pyswmm/raingages.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        >>>
+        >>> with Simulation('tests/data/TestModel1_weirSetting.inp') as sim:
+        ...     rg = RainGage(sim)["Gage1"]
+        ...     print(rg.raingageid)
+        >>> Gage1
+        """
+        return self._raingageid
+
+    @property
+    def total_precip(self):
```
Comment (jennwuu) [inline pyswmm/raingages.py]: Does this change the precipitation rate of rain gage? total_precip sounds confusing to me for a rate (in/hr or mm/hr)

Comment (bemcdonnell) [reply pyswmm/raingages.py]: @jennwuu, in my imagination we were going to hope people would read the documentation to know that this is the total precipitation rate (not cumulative).  Agree that the name is a little misleading, however total_precip = rainfall + snowfall rates

File: pyswmm/tests/data/__init__.py (hunk 1 above)
Comment (bemcdonnell) [reply pyswmm/tests/data/__init__.py]: Ill re-org this a little

File: pyswmm/tests/test_rainfallAPI.py (hunk 2 above)
Comment (bemcdonnell) [reply pyswmm/tests/test_rainfallAPI.py]: Yes, we fan inject our own hyetograph during the simulation ;-) that's why this feature is baller.

File: pyswmm/raingages.py (hunk 3 above)
Comment (jennwuu) [reply pyswmm/raingages.py]: would it make sense to put total_precip_rate or is that too long?

Comment (bemcdonnell) [reply pyswmm/raingages.py]: @jennwuu, I think it's tooooo expressive to add that.

**Pull request merged**